asl/encoding/encoding.go

```diff
@@ -45,6 +48,16 @@ func AppendUint32ToBufferLE(buffer []byte, v uint32) []byte {
 	return binary.LittleEndian.AppendUint32(buffer, v)
 }
 
+func AppendUint32ToBufferLEAsVarInt(buffer []byte, v uint32) []byte {
```

Comment:
I don't think there is any need for these methods. There's already a binary.AppendUvarint() method which does this, and does not do extra allocations like this method does. Here's an example of its use https://github.com/spirit-labs/tektite/blob/main/kafkaprotocol/api_versions_request.go#L64

Reply:
Thanx for the pointer. Was not aware let me check those section.

Reply:
Removed the helpers and using the package methods directly